synapse/storage/room.py

```diff
@@ -541,7 +543,14 @@ def _store_content_index_txn(self, txn, event, key):
     def _update_retention_policy_for_room_txn(self, txn, event):
         if (
             hasattr(event, "content")
-            and ("min_lifetime" in event.content or "max_lifetime" in event.content)
+            and (
+                ("min_lifetime" in event.content and isinstance(
+                    event.content.get("min_lifetime"), integer_types
+                ))
+                or ("max_lifetime" in event.content and isinstance(
+                    event.content.get("max_lifetime"), integer_types
+                ))
+            )
         ):
             self._simple_update_one_txn(
```

Comment:
We should probably insert rather than update, if we're going to do the join-on-current-state-events trick.